presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergPageSourceProvider.java

```diff
@@ -0,0 +1,667 @@
+                            Optional.empty(),
+                            Optional.empty()));
+                }
+            }
+
+            TupleDomain<HiveColumnHandle> hiveColumnHandleTupleDomain = effectivePredicate.transform(column -> {
+                IcebergOrcColumn icebergOrcColumn;
+                if (fileOrcColumnByIcebergId.isEmpty()) {
+                    icebergOrcColumn = fileOrcColumnsByName.get(column.getName());
+                }
+                else {
+                    icebergOrcColumn = fileOrcColumnByIcebergId.get(column.getId());
+                    if (icebergOrcColumn == null) {
+                        // Cannot get orc column from 'fileOrcColumnByIcebergId', which means SchemaEvolution may have happened, so we get orc column by column name.
+                        icebergOrcColumn = fileOrcColumnsByName.get(column.getName());
+                    }
+                }
+
+                return new HiveColumnHandle(
+                        column.getName(),
+                        toHiveType(column.getType()),
+                        column.getType().getTypeSignature(),
+                        // Note: the HiveColumnHandle.hiveColumnIndex starts from '0' while the IcebergColumnHandle.id starts from '1'
+                        icebergOrcColumn != null ? icebergOrcColumn.getOrcColumnId() : column.getId() - 1,
+                        icebergOrcColumn != null ? icebergOrcColumn.getColumnType() : REGULAR,
+                        Optional.empty(),
+                        Optional.empty());
+            });
+
+            OrcPredicate predicate = new TupleDomainOrcPredicate<>(hiveColumnHandleTupleDomain, columnReferences.build(), orcBloomFiltersEnabled, Optional.of(domainCompactionThreshold));
+
+            OrcAggregatedMemoryContext systemMemoryUsage = new HiveOrcAggregatedMemoryContext();
+            OrcBatchRecordReader recordReader = reader.createBatchRecordReader(
+                    includedColumns.build(),
+                    predicate,
+                    start,
+                    length,
+                    UTC,
+                    systemMemoryUsage,
+                    INITIAL_BATCH_SIZE);
+
+            return new OrcBatchPageSource(
+                    recordReader,
+                    orcDataSource,
+                    physicalColumnHandles,
+                    typeManager,
+                    systemMemoryUsage,
+                    stats,
+                    runtimeStats);
+        }
+        catch (Exception e) {
+            if (orcDataSource != null) {
+                try {
+                    orcDataSource.close();
+                }
+                catch (IOException ignored) {
+                }
+            }
+            if (e instanceof PrestoException) {
+                throw (PrestoException) e;
+            }
+            String message = format("Error opening Iceberg split %s (offset=%s, length=%s): %s", path, start, length, e.getMessage());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: start (declared at line 427), length (declared at line 428). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 595. Consider formatting this data with &#39;%d&#39; instead.